spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/BlockingQueueConsumer.java

```diff
@@ -661,6 +661,9 @@ public void stop() {
 		}
 		RabbitUtils.setPhysicalCloseRequired(true);
 		ConnectionFactoryUtils.releaseResources(this.resourceHolder);
+		if (!(this.channel instanceof ChannelProxy)) { // clear the TL flag if the channel is not a proxy
+			RabbitUtils.isPhysicalCloseRequired();
```

Comment:
We may consider just don't set TL flag if `channel` isn't `Proxy`.
See a couple lines above - `RabbitUtils.setPhysicalCloseRequired(true)`.